### DISCUSSION TITLE
Add a command to list all valid targets for a given project

### DISCUSSION THREAD

**Discussion by github-tma8118** [nrwl/nx#22100] [Feature Requests]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

I'd like to have an option to list all the valid targets for a given project.

For example, similar to how `nx print-affected --select=projects` prints a comma-separated list of affected projects, I'd like to do something like `nx print-targets --project=some-affected-project` to print a comma-separated list of valid targets defined in that project.

## Motivation

I recently tried to implement a post-commit hook to print some helpful information for users after they made a commit.

The hook prints some commands to the terminal to encourage users to run things like unit tests locally before pushing them and consuming billable pipeline minutes just for the unit tests to fail, meaning that the user has to fix the tests and push again and consume more minutes.

I was able to get a list of affected projects and, for each one, suggest `nx run some-affected-project:test`, but not all of my projects have a `test` target, so for some projects that suggestion doesn't make sense.

It would be nice to be able to print the valid targets of each project.

Other tools such as Gradle have a similar concept, for example `gradle tasks` lists valid tasks for the current project:

![Gradle-Tasks-9-284594423](https://github.com/nrwl/nx/assets/109232127/ad395073-8e23-4a46-8567-bba10fdd76d3)

Although a comma-separated list would be easier to consume and iterate over in something like a post-commit hook.

**Comment by jaysoo** [comment]
You can currently get project data with the `nx show project <name> --json` command.

For example, for a project called `demo`, you can run:

```shell
npx nx show project demo --json | jq -r '.targets | to_entries[] | .key'
```

Which would show all targets, such as:

```
test
lint
build
serve
```

And you can use that to script your post-commit hook.

**Reply by Fa7** [reply]
I just ran the following with a helper script to pass a target 

`npx nx show projects --with-target $0`

It will print something like
```
react-native
someLibrary
@SomeLib/LibName
```

This will list all projects associated with a given target.
